docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,188 @@
+
+Typically this will include
+1. code at the stimulus/event routing layer (e.g. http routing, message handlers)
+1. factories to be used by the client code to create tenant-specific instances of service clients, configured appropriately
+
+## Solution-layer concerns
+As we dig down in the stack, we finally reach implementation concerns in the multi-tenanted solution itself.
+
+How do you design a SQL Azure database to host multi-tenanted data? What are the isolation characteristics? What about Blob storage? Cosmos DB? 
+
+These solution-level design choices are well explored in the Azure documentation for each individual service. For example
+- [Azure SQL Database]( https://docs.microsoft.com/en-us/azure/azure-sql/database/saas-tenancy-app-design-patterns)
+- [Cosmos DB]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/cosmos-db)
+- [Azure Storage]( https://docs.microsoft.com/en-us/azure/architecture/guide/multitenant/service/storage)
+
+When reading that documentation, it is important to consider whether it describes multi-tenant or multi-user scenarios, as we have defined them.
+They sometimes use terms such as *hostile* and *non-hostile* tenants (respectively), to distinguish between these models.
+
+In general, the Azure documentation describes isolation mechanisms for a particular compute or storage service, to support different tenancy and user isolation models, rather than considering a more holistic/solution view of tenancy, bringing together a variety of services into a multi-tenanted solution.
+
+It gives you the nuts-and-bolts (we will talk more about this below), but you will need to understand how each service you consume slots into your overall tenancy model.
+
+For any given technology or service, there are implementation choices which trade-off a number of factors:
+
+1. Security
+    - Does a breach of this service or malicious code execution in one tenant offer a mechanism to hop-over into another tenant?
+1. Noisy neighbours
+    - Does the operation of one tenant have performance or scalability implications for the operation of another tenant? Can one tenant deny or degrade service to others?
+1. Scalability and density
+    - What are the scalability characteristics of the solution as an individual tenant scales up, and/or the number of tenants scales up?
+1. Portability
+    - Can a tenant be moved (transparently) from one physical resource to another to resolve conflicts and/or as the tenant scales/buys more resources?
+1. Cost
+    - What is the cost model for the resource?
+1. Deployment and Manageability
+    - What are the deployment and manageability implications as the number of tenants scales up? What are the observability and monitoring characteristics for a an individual tenant? For the root tenant?
+1. Versioning and updates
+    - Are there implications for versioning or updating tenanted solutions? What about the cost of updating the management plane for those services?
```

Comment:
Is that (independent) versioning of tenants?

Reply:
Yes; a bit more added there.